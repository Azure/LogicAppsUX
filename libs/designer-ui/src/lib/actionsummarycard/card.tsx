import { Text, Image, ImageFit } from '@fluentui/react';
import { getIntl } from '@microsoft-logic-apps/intl';

export type OperationCardProps = {
  title: string;
  subtitle?: string;
  id: string;
  iconUrl: string;
  connectorName: string;
  category: 'Built-in' | 'Azure' | '';
} & CommonCardProps;

export interface CommonCardProps {
  brandColor?: string;
}

export const OperationCard = (props: OperationCardProps) => {
  const intl = getIntl();
  const logoAltText = intl.formatMessage(
    {
      defaultMessage: 'logo for {connectorName}',
      description: 'Alternate text for accessibility, describes logo for corresponding connector',
    },
    {
      connectorName: props.connectorName,
    }
  );

  const triggerFilterText = intl.formatMessage({
    defaultMessage: 'Trigger',
    description: 'Filter that selects only Trigger Operations',
  });

  return (
    <div className="msla-operation-card">
      <div>
        <div className="msla-card-title-container">
          <Image imageFit={ImageFit.contain} className="msla-card-logo" src={props.iconUrl} alt={logoAltText}></Image>
        </div>
        <div className="msla-card-title-container">
          <Text className="msla-card-title">{props.title}</Text>
        </div>
      </div>
      <Text className="msla-card-description">{props.subtitle}</Text>
      <div className="msla-tag-container">
        <Text className="msla-tag">{props.connectorName}</Text>
        <Text className="msla-tag">{triggerFilterText}</Text>
        <Text className="msla-tag">{props.category}</Text>
      </div>
    </div>
  );
};
