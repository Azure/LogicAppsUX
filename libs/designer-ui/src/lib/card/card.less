@import (reference) '../variables.less';
@import './config/config.less';
@import './connectiontypeselector/connectiontypeselector.less';
@import '../selector/listitem.less';
@import './operationconfigurations/operationconfigurations.less';
@import '../flyout/flyout.less';
@import './parameters/parameters.less';
@import './function/function.less';
@import './scope/scope.less';

.msla-card-loading {
  font-family: @font-family;
  margin: 10px;
  padding: 10px;
}

.msla-component-error {
  & > span {
    color: rgb(224, 2, 2);
    font-size: 14px;
    vertical-align: top;
  }
}

.card-narrow-width-mixin() {
  /* For mobile the min width cannot be the maximum width */
  min-width: @card-min-width;
  max-width: @card-max-width;
  width: 100%;

  @media (-ms-high-contrast: active), (-ms-high-contrast: none) {
    box-sizing: border-box;
    min-width: @card-min-fixed-width;
  }
}

.msla-card,
.msla-card-placeholder {
  &.msla-card-fixed-width {
    margin: 0 auto;
    min-width: @card-min-fixed-width;
    width: @card-max-width;

    @media only screen and (max-width: @designer-desktop-max-width) {
      .card-narrow-width-mixin();
    }

    &.msla-no-triggers {
      font-family: @font-family;
      font-size: @card-header-font-size;
      padding: 0.5em;
      text-align: center;
    }
  }

  &.msla-card-variable-width {
    display: table;
    margin: 0 auto;
    min-width: @card-max-width;

    @media only screen and (max-width: @designer-desktop-max-width) {
      .card-narrow-width-mixin();
    }
  }
}

.msla-connectionwizard-canvas--narrow {
  .msla-card,
  .msla-card-placeholder {
    &.msla-card-fixed-width {
      .card-narrow-width-mixin();
    }

    &.msla-card-variable-width {
      .card-narrow-width-mixin();
    }
  }
}

.msla-card,
.msla-card-placeholder {
  margin: 0 auto;
}

.msla-card-container,
.msla-card-container-cell {
  margin: 0 auto;
  box-sizing: border-box;
  position: relative;
  vertical-align: top;
  width: 100%;

  /* Workaround for IE-only CSS bug */
  @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
    width: auto;
  }
}

.msla-card-container-empty {
  width: 0;
  display: inline-block;
}

.msla-card-view {
  box-sizing: border-box;
  margin: 0 auto;
  width: 100%;
}

.msla-card-caption {
  padding: 0 3px;
  .msla-drop-zone > .msla-connector-add-step {
    left: @insert-button-offset;
  }
}

.msla-card-children {
  display: flex;
  text-align: center;
}

.msla-diet-view .msla-card-children {
  text-align: left;
}

.msla-card-recommendation-placeholder {
  background-color: '#FFF';
}

.msla-card,
.msla-card-placeholder {
  min-width: 200px;
  border: @defaultBorder;
  box-sizing: content-box;
  font-family: @font-family;
  font-size: @card-body-font-size;

  .msla-card-body {
    text-align: justify;

    .msla-card-inner-body {
      padding: 12px;

      .msla-card-section-title {
        font-size: 16px;
      }
    }

    .msla-recommendation-v3 {
      .msla-connectors {
        padding: 12px 7px 3px 7px;
      }
    }
  }

  .msla-card-header.msla-dark-header {
    .msla-card-title-group {
      .msla-card-header-title-view {
        color: #f7f7f7;
      }
    }
  }

  .msla-card-header {
    font-family: @header-font-family;
    font-size: @card-header-font-size;
    text-align: left;
    height: @card-header-height;
    line-height: @card-header-height;
    color: @card-text-color;

    display: flex;
    justify-content: space-between;

    .msla-card-title-group {
      display: flex;
      overflow: hidden;
      flex: 1 1 auto;
      align-items: center;
      cursor: pointer;

      /* Workaround for IE-only flexbox bug */
      @media only screen {
        @media (-ms-high-contrast: active), (-ms-high-contrast: none) {
          flex: 1 1 100%;
        }
      }

      .msla-card-header-title-view,
      .msla-card-header-title-edit {
        display: flex;
        align-items: center;
      }

      .msla-card-header-title-edit {
        flex: 1 1 auto;
      }

      .msla-card-header-title-view {
        cursor: pointer;
        color: @card-text-color;
        overflow: hidden;
        outline: none;
        text-overflow: ellipsis;
        white-space: nowrap;
        height: inherit;
        padding-left: 10px;
        display: flex;

        &:link,
        &:visited {
          text-decoration: none;
        }

        .msla-card-header-title-tag {
          color: #646464;
          padding-left: 4px;
        }
      }

      .msla-card-header-title-edit {
        height: inherit;
      }

      .msla-card-header-logo {
        width: @card-header-height;
        height: @card-header-height;
        display: inline-block;
        vertical-align: top;
        cursor: pointer;
        flex: 0 0 auto;

        .msla-card-header-icon {
          width: 100%;
          height: 100%;
        }
      }

      .msla-card-title {
        text-overflow: ellipsis;
        overflow: hidden;
      }

      .msla-ise {
        border: @badge-border solid #666;
        color: #666;
        display: inline-block;
        font-size: 11px;
        height: 14px;
        line-height: 13px;
        margin-left: @badge-margin-left;
        margin-right: @badge-margin-right;
        text-align: center;
        text-transform: uppercase;
        min-width: @badge-width;
        max-width: @badge-width;
      }
    }

    &.msla-component-error {
      display: block;

      & > span {
        color: rgb(224, 2, 2);
        vertical-align: top;
      }
    }
  }

  .msla-iframe-terms-of-service {
    padding-top: 10px;
  }

  .msla-iframe-terms-of-service iframe {
    border: 1px solid @defaultBorderColor;
    width: 100%;
    height: 200px;
    min-width: calc(100% - 20px);
    background: white;
  }
}

.msla-card-selected {
  box-shadow: 0 0 12px @selectedShadowColor;
  border-width: 2px;
}

.msla-card,
.msla-card-placeholder,
.msla-recommendation-dropdown,
.msla-recommendation-footer,
.msla-recommendation-empty {
  background-color: @card-background-color;
}

.msla-recommendation-container {
  text-align: justify;
}

.msla-card-auth-body {
  padding: 10px;
  text-align: center;

  .msla-card-button-primary,
  .msla-card-button-secondary {
    margin-top: 10px;
    margin-right: 10px;
    min-width: 150px;
  }

  .msla-card-config-item {
    text-align: left;

    > label {
      flex-basis: 24%;
      margin-right: 0;
    }
  }

  .msla-auth-options {
    display: flex;
    flex-direction: row;
  }

  .errorMessage-managed-identity {
    color: #a4262c;
    text-align: left;
    margin-bottom: 10px;
    margin-left: 27%;
    @width-offset: (@parameter-label-width + @parameter-label-inputbox-margin);
    max-width: calc(100% ~'-' @width-offset);
  }
}

.msla-card-auth-description {
  margin: 0;
  padding-bottom: 10px;
  word-wrap: break-word;
}

.msla-card-button-primary {
  line-height: 30px;
  padding: 0 40px;
}

.msla-card-button-secondary {
  line-height: 30px;
  padding: 0 40px;
}

.msla-card-profile-body {
  padding: 0 0 12px 0;
}

.msla-card-profile-description {
  line-height: @parameter-inputbox-height;
  margin: 0;
}

.msla-card-profile-list {
  max-height: 380px;
  /* 10 items */
  overflow-y: auto;
}

.msla-card-profile-buttons {
  text-align: center;

  [class^='ms-Button']:not(:last-child) {
    margin-right: 4px;
  }
}

.msla-card-swagger-body {
  padding: @parameter-label-inputbox-margin;
}

.msla-swagger-endpoint-label {
  text-transform: uppercase;
  color: @card-text-color;
}

.msla-swagger-endpoint-box {
  border: 1px solid #989898;
}

.msla-swagger-endpoint-confirm {
  padding-top: 12px;
  text-align: center;
}

.msla-function-inputs-footer {
  margin-top: @parameter-interval-margin;
  padding-bottom: 12px;
  text-align: center;
}

.msla-card-title-button-group {
  display: flex;
  align-items: center;
  flex: 0 0 auto;
}

.msla-card-title-button {
  height: @card-header-height;
  width: @card-header-height;
  float: left;
  /* float:left is needed to workaround for IE-only bug */
  color: #323130;

  .msla-card-title-button-icon {
    vertical-align: middle;
    font-size: calc(@card-header-height / 2);
  }
}

.msla-card-title-button-static-result {
  color: @static-result-icon-color;
}

.msla-card-title-img {
  height: calc(@card-header-height / 2);
  width: @card-header-height;
  float: left;
}

.msla-connector-top {
  border-top: 2px solid @defaultConnectorColor;
}

.msla-step-connector-arrow-down {
  display: table;
  width: 100%;
  margin: auto;
  height: @card-connector-default-height;

  & > div {
    display: table-row;
    height: inherit;

    & > div {
      display: table-cell;
    }
  }
}

.msla-step-connector-info-section {
  position: absolute;
  left: calc(50% - 55px);
  top: 0;
  z-index: 1;

  .msla-step-connector-info-button {
    height: @card-connector-default-height;
    width: 40px;
    position: absolute;
    top: 0;
    color: black;
  }
}

.msla-connector-add-step {
  height: inherit;
  position: absolute;
  left: @insert-button-offset;
  top: 4px;
  z-index: 1;

  &.msla-connector-arrow-left,
  &.msla-connector-arrow-right {
    height: @card-connector-default-height;
  }

  &.viewmanager2 {
    position: static;
    left: auto;
    top: auto;

    .msla-connector-add-step-button {
      position: relative;
      left: 1px;
    }
  }
}

.msla-connector-add-step-button:focus {
  outline: 2px dotted #000;
}

.msla-main-page {
  .msla-connector-add-step-button:focus {
    outline: none;
  }
}

.msla-step-connector-arrow-right,
.msla-step-connector-arrow-left,
.msla-step-connector-join,
.msla-step-connector-arrow-both {
  display: table;
  width: 100%;
  height: @card-connector-default-height;
  min-height: @card-connector-default-height;

  & > div {
    display: table-row;
    height: 100%;

    & > div {
      display: table-cell;

      &.msla-connector-arrow {
        width: 50%;
        box-sizing: border-box;

        .msla-connector-top-line {
          border-top: 2px solid @defaultConnectorColor;
        }
      }

      &.msla-connector-arrow-right {
        width: 100%;
        box-sizing: border-box;
        position: relative;

        &.background {
          padding-left: 50%;
        }

        .msla-connector-top-line {
          top: 0;
          width: 100%;
          border-top: 2px solid @defaultConnectorColor;
        }
      }

      &.msla-connector-arrow-left {
        width: 100%;
        box-sizing: border-box;
        position: relative;

        &.background {
          padding-right: 50%;
        }

        .msla-connector-top-line {
          top: 0;
          width: 100%;
          border-top: 2px solid @defaultConnectorColor;
        }
      }

      &.msla-connector-right {
        width: 50%;
        box-sizing: border-box;

        &.background {
          padding-left: 9px;
        }

        .msla-connector-top-line {
          border-top: 2px solid @defaultConnectorColor;
        }
      }

      &.msla-connector-left {
        width: 50%;
        box-sizing: border-box;

        &.background {
          padding-right: 9px;
        }

        .msla-connector-top-line {
          border-top: 2px solid @defaultConnectorColor;
        }
      }
    }
  }
}

.msla-card-flex {
  display: flex;
  flex-direction: column;
  height: 100%;
  width: auto;
}

.msla-card-view-div {
  height: auto;
}

.msla-card-connector-flex {
  box-sizing: border-box;
  flex: 1;
  height: 100%;
  min-height: @card-connector-default-height;
}

.msla-connector-vertical-flex {
  flex: 1 0 30px;
}

.msla-step-connector-bottom-left,
.msla-step-connector-bottom-right,
.msla-step-connector-bottom-both,
.msla-step-connector-vertical {
  width: 100%;
  height: 100%;
  position: relative;

  & > div {
    display: flex;
    height: 100%;

    & > div {
      flex: 1;

      &.msla-connector-vertical-line {
        border-right: 2px solid @defaultConnectorColor;
        min-height: 30px;
      }

      &.msla-connector-bottom-right-line {
        border-left: 2px solid @defaultConnectorColor;
        border-bottom: 2px solid @defaultConnectorColor;
      }

      &.msla-connector-bottom-left-line {
        border-right: 2px solid @defaultConnectorColor;
        border-bottom: 2px solid @defaultConnectorColor;
      }

      &.msla-connector-bottom-line {
        border-bottom: 2px solid @defaultConnectorColor;
      }
    }
  }
}

.msla-step-connector-dummy-div {
  height: 0;
  width: 0;
}

.msla-node-margin {
  margin-bottom: 20px;
}

/* Style still in use that used to be in informationbox.less which was removed */
.msla-information-box-container {
  .msla-information-box {
    font-size: 12px;
    padding: 12px;
    right: 25px;
    width: 248px;

    a {
      &:active,
      &:link,
      &:hover,
      &:visited {
        text-decoration: underline;
        color: @brandColor;
      }
    }
  }
}

.msla-service-principal-div,
.msla-managed-identity-div,
.msla-sign-in-div {
  text-align: left;
  margin-top: 10px;
  margin-right: 5px;

  button {
    display: inline-block;
  }

  .msla-button {
    line-height: 28px;
    img {
      width: 15px;
      height: 15px;
      vertical-align: sub;
      cursor: pointer;
    }
  }
}

.msla-designer-canvas {
  .msla-card-body {
    padding: 5px 5px 10px 5px;
  }
}

.msla-badge {
  border: 1px solid #666;
  color: #666;
  display: inline-block;
  font-size: 11px;
  height: 14px;
  line-height: 13px;
  text-align: center;
  text-transform: uppercase;
  padding: 1px 4px;
  margin-left: 6px;
}

body.dark {
  .msla-connector-add-step-button {
    &:focus {
      outline: 2px dotted @ms-color-black;
    }
  }

  .msla-main-page {
    .msla-connector-add-step-button:focus {
      outline: none;
    }
  }

  .msla-card,
  .msla-card-placeholder {
    background-color: @ms-color-white;
  }

  .msla-main-page {
    .msla-card,
    .msla-card-placeholder {
      background-color: #252423;
    }
  }

  .msla-tab-component-container {
    .msla-card,
    .msla-card-placeolder {
      background-color: inherit;
    }
  }

  .msla-card-header {
    color: @ms-color-black;

    .msla-card-title-group {
      .msla-card-header-title-view {
        color: @ms-color-black;

        .msla-card-header-title-tag {
          color: @ms-color-neutralTertiary;
        }
      }

      .msla-ise {
        border-color: @ms-color-neutralTertiary;
        color: @ms-color-neutralTertiary;
      }
    }
  }

  .msla-badge {
    border-color: @ms-color-neutralTertiary;
    color: @ms-color-neutralTertiary;
  }

  .msla-badges {
    border-top: 1px solid #e1dfdd;

    .panel-card-v2-badge {
      &.active {
        color: @ms-color-neutralTertiary;

        &:focus {
          outline: 1px solid #a19f9d;
        }
        &:hover {
          background-color: @ms-color-neutralLighter;
        }
        &:active {
          background-color: @ms-color-neutralLight;
        }
      }

      &.inactive {
        color: #a19f9d;
      }
    }
  }

  .msla-panel-card-container {
    .panel-card-main {
      background-color: #2f2f2f;

      .panel-card-error {
        &.error,
        &.blocked {
          color: #f1707b;
        }

        &.severeWarning {
          color: #fce100;
        }

        &.success {
          color: #92c353;
        }

        &.info,
        &.warning {
          color: #a19f9d;
        }
      }
    }
  }

  .errorMessage-managed-identity {
    color: #f1707b;
    .managed-identity-link {
      color: #69afe5;
    }
  }
}

.msla-panel-card-container {
  position: relative;
  border: 1px solid #c8c6c4;
  box-sizing: border-box;
  box-shadow: 0 0.3px 0.9px rgba(0, 0, 0, 0.1), 0 1.6px 3.6px rgba(0, 0, 0, 0.13);
  font-size: 12px;
  display: inline-block;
  border-radius: 2px;
  width: @panel-mode-card-width;
  -webkit-user-select: none;
  -ms-user-select: none;
  user-select: none;

  &:focus {
    outline: 2px solid #605e5c;
  }

  &:focus,
  &:hover {
    box-shadow: 0 1.2px 3.6px rgba(0, 0, 0, 0.1), 0 6.4px 14.4px rgba(0, 0, 0, 0.13);

    .panel-card-content-gripper-section {
      visibility: visible;
      cursor: grab;
    }
  }

  &.msla-panel-card-container-selected {
    border: 2px solid #0078d4;
    box-shadow: 0 0.3px 0.9px rgba(0, 0, 0, 0.1), 0 1.6px 3.6px rgba(0, 0, 0, 0.13);
  }

  &.dragging {
    cursor: grabbing;
    opacity: 0.7;
    outline: none;
    box-shadow: 0 1.2px 3.6px rgba(0, 0, 0, 0.1), 0 6.4px 14.4px rgba(0, 0, 0, 0.13);

    .panel-card-content-gripper-section {
      visibility: hidden;
    }
  }

  &.msla-card-ghost-image {
    position: fixed;
    pointer-events: none;
    opacity: 100%;
    z-index: 12;
    box-shadow: 0 1.2px 3.6px rgba(0, 0, 0, 0.1), 0px 12px 40px rgba(0, 0, 0, 0.24);

    .panel-card-content-gripper-section {
      visibility: visible;
    }
  }

  .panel-card-main {
    display: flex;
    flex-direction: column;
    align-items: stretch;
    cursor: pointer;
    background-color: #fff;

    .panel-card-header {
      .panel-card-content-container {
        flex: 1 1 auto;
        display: flex;
        flex-direction: row;
        min-height: 32px;

        .panel-card-right-controls {
          display: flex;
          flex-direction: column;

          @media (-ms-high-contrast: active), (-ms-high-contrast: none) {
            margin-top: 5px;
          }

          .msla-card-edit-button {
            overflow: hidden;
            flex: 0 0 30px;
            margin-top: 9px;
          }
        }

        .panel-card-content-icon-section {
          flex: 0 0;
          margin: 8px 8px 0 5px;

          @media (-ms-high-contrast: active), (-ms-high-contrast: none) {
            flex-basis: auto;
          }

          .panel-card-icon {
            height: 24px;
            width: 24px;
            border-radius: 2px;
          }
        }

        .panel-card-top-content {
          align-self: center;
          flex: 1 1 auto;
          font-size: 13px;
          display: flex;
          justify-content: space-between;

          .panel-msla-title {
            align-self: center;
            display: inline-block;
            font: 14px/20px normal @semibold-font-family;
            margin: 10px 8px 10px 0;
            text-align: left;
            word-break: break-word;
          }

          .msla-card-title {
            cursor: pointer;
            color: @card-text-color;
            text-align: left;
          }
        }

        .panel-card-expand {
          flex: 0 0 8px;
          display: inline-block;
          vertical-align: middle;
          background-color: transparent;
          border: none;

          @media (-ms-high-contrast: active), (-ms-high-contrast: none) {
            flex: 0 0 auto;
          }

          &.loading {
            padding-bottom: 10px;
          }
        }
      }
    }

    .panel-card-error {
      margin: -2px 10px 8px 12px;
      text-overflow: ellipsis;
      overflow: hidden;
      text-align: left;
      white-space: nowrap;

      &.error,
      &.blocked {
        color: #a80000;
      }

      &.severeWarning {
        color: #d83b01;
      }

      &.success {
        color: #107c10;
      }

      &.info,
      &.warning {
        color: #605e5c;
      }
    }
  }

  .panel-card-content-gripper-section {
    margin: 12px 0 0 5px;
    visibility: hidden;
  }

  .panel-card-bottom-toolbar {
    flex: 0 0 30px;
    height: 35px;
    width: 100%;
  }

  .panel-card-slideout {
    position: absolute;
    right: 0;
    top: 0;
    z-index: 100;

    .panel-card-badges-container {
      padding-left: 10px;
      flex: 0 0 30px;
      display: flex;
      font-size: 12px;
      justify-content: space-between;

      .panel-card-badges-section {
        display: flex;
        margin-right: 5px;
      }

      .panel-badge-icon {
        align-self: center;
        justify-self: flex-end;
      }
    }

    .panel-card-slideout-content {
      box-shadow: 0 0 12px rgba(0, 0, 0, 0.35);
      width: 350px;
      position: absolute;
      top: -2px;
      display: flex;
      flex-direction: column;
      justify-content: center;
      overflow-y: auto;
      background-color: white;

      .panel-card-fields-container {
        display: flex;
        flex-direction: column;
        padding-bottom: 10px;

        .panel-card-field {
          display: flex;
          flex-direction: column;
          margin: 0 5px 0 5px;
          text-align: left;

          .panel-card-field-title {
            padding-left: 7px;
            margin: 10px 0 5px 0;
          }

          .panel-card-field-value {
            padding-left: 7px;
          }
        }
      }
    }
  }

  .panel-card-badge {
    align-items: center;
    display: flex;
    font-size: 11px;
    padding: 0 4px;
    margin: 5px;
    border: 1px solid rgb(100, 100, 100);
  }
}

.msla-main-page {
  .msla-panel-card-container {
    &.inactive {
      opacity: 0.5;

      .panel-card-main {
        cursor: default;
      }
    }
  }
}

.msla-tab-component-container,
.msla-tab-component-container > .msla-if {
  .msla-card:not(.msla-run-after-logo) {
    border: none;
    box-shadow: none;

    > .msla-card-header {
      height: 0;
      overflow: hidden;
      visibility: hidden;
    }
  }
}

.msla-badges {
  border-top: 1px solid #e1dfdd;
  display: flex;
  justify-content: flex-end;
  padding: 2px 5px;
  text-align: left;

  &:empty {
    display: none;
  }

  .panel-card-v2-badge {
    padding: 4px;
    font-size: 12px;

    &.active {
      color: #323130;

      &:focus {
        outline: 1px solid #605e5c;
      }
      &:hover {
        background-color: #f3f2f1;
      }
      &:active {
        background-color: #edebe9;
      }

      &.darkBackground {
        color: #fff;

        &:focus {
          outline: 1px solid #fff;
        }
        &:hover {
          background-color: rgba(0, 0, 0, 0.2);
        }
      }
    }

    &.inactive {
      color: #a19f9d;
    }
  }
}

body.dark {
  .msla-information-box-container {
    .msla-information-box {
      a {
        &:active,
        &:link,
        &:hover,
        &:visited {
          color: @ms-color-themePrimary;
        }
      }
    }
  }

  .msla-panel-card-container {
    &:focus {
      outline-color: #babab5;
    }

    &:hover {
      .panel-card-main {
        background-color: #484644;
      }
    }

    .panel-card-slideout {
      .panel-card-slideout-content {
        background-color: @ms-color-white;
        color: @ms-color-black;
      }
    }
  }

  .msla-button-link {
    color: @ms-color-themePrimary;
  }

  .msla-card-button-primary {
    background-color: @ms-color-themePrimary;
    color: @ms-color-white;
  }

  .panel-card-content-container {
    .panel-card-content-gripper-section {
      svg path {
        fill: #c8c6c4;
      }
    }
  }
}

// NOTE(nidurak): diet view related overrides must come last to take precedence
.msla-diet-view {
  .msla-card-container {
    min-width: @diet-view-card-min-width;
  }

  .msla-card-caption {
    padding: 0;
    float: left;
    display: block;
  }
}
