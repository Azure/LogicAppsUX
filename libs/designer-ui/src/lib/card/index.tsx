export const NYI = 'NYI';
export * from './cardv2';
