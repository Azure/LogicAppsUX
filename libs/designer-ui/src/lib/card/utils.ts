import Constants from '../constants';
import type { IButtonStyles } from '@fluentui/react';
import { hexToRgbA } from '@microsoft/logic-apps-shared';

export function getCardStyle(brandColor?: string): React.CSSProperties {
  return {
    borderLeft: `4px solid ${brandColor}`,
    borderRadius: '2px',
  };
}

export function getCardButtonsStyle(themeColor: string): IButtonStyles {
  return {
    icon: {
      color: themeColor,
      width: 15,
      height: 15,
    },
    flexContainer: {
      width: 15,
      height: 15,
    },
    root: {
      marginTop: 11,
      padding: 0,
      margin: 0,
      width: 15,
      marginLeft: 12,
      marginRight: 10,
    },
  };
}

export function getHeaderStyle(brandColor?: string): React.CSSProperties | undefined {
  return brandColor ? { backgroundColor: brandColor } : undefined;
}

// This is often unnecessary and takes ~0.5ms to run each time
// Not an issue on small uses but with 500 operations it's 250ms * each use to run
export function getBrandColorRgbA(brandColor = Constants.DEFAULT_BRAND_COLOR, opacity = 0.15): string {
  try {
    return hexToRgbA(brandColor, opacity);
  } catch {
    return hexToRgbA(Constants.DEFAULT_BRAND_COLOR, opacity);
  }
}

const opacityHexValues = {
  1.0: 'FF',
  0.7: 'B3',
  0.5: '80',
  0.3: '4D',
  0.15: '26',
  0.1: '1A',
};

// Faster brand color with opacity, no need to calculate, we can just use common values
export function getBrandColorWithOpacity(brandColor = Constants.DEFAULT_BRAND_COLOR, opacity: keyof typeof opacityHexValues): string {
  return `${brandColor}${opacityHexValues[opacity]}`;
}
