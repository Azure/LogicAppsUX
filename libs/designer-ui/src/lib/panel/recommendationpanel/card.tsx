import { Text, Image } from '@fluentui/react';

export type OperationCardProps = {
  title: string;
  subtitle?: string;
  id: string;
  iconUrl: string;
} & CommonCardProps;

export interface CommonCardProps {
  brandColor?: string;
}

export const OperationCard = (props: OperationCardProps) => {
  return (
    <div key={props.id} className="msla-operation-card">
      <Image src={props.iconUrl} alt={'logo for ' + props.title}></Image>
      <Text>{props.title}</Text>
      <Text>{props.subtitle}</Text>
    </div>
  );
};
