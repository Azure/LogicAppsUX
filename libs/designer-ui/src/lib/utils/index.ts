export * from './keyboardUtils';
export * from './theme';
export * from './utils';
