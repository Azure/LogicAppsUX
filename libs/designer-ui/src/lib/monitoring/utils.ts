import { getDurationString } from '../utils';

export function calculateDuration(startTime: string, endTime: string | undefined): string {
  return endTime ? getDurationString(Date.parse(endTime) - Date.parse(startTime), /* abbreviated */ false) : getDurationString(Number.NaN);
}
