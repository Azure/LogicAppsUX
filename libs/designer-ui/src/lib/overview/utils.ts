import { getDurationString } from '../utils';
import type { RunDisplayItem } from './types';
import type { Run, RunError } from '@microsoft/utils-logic-apps';
import { isObject } from '@microsoft/utils-logic-apps';

export function isRunError(value: any): value is RunError {
  return isObject(value) && isObject(value.error) && typeof value.error.code === 'string' && typeof value.error.message === 'string';
}

export function mapToRunItem({ id, name: identifier, properties }: Run): RunDisplayItem {
  const { endTime, startTime, status } = properties;
  const duration = endTime ? getDurationString(Date.parse(endTime) - Date.parse(startTime), /* abbreviated */ false) : '--';

  return {
    duration,
    id,
    identifier,
    startTime,
    status,
  };
}
