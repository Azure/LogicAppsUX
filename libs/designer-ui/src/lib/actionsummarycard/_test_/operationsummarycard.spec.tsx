import type { OperationCardProps } from '../card';
import { OperationCard } from '../card';
import { MockSearchOperations } from '@microsoft-logic-apps/utils';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('recommendation panel', () => {
  let renderer: ShallowRenderer.ShallowRenderer;
  const operation = MockSearchOperations[0];
  const { properties } = operation;

  beforeAll(() => {
    renderer = ShallowRenderer.createRenderer();
  });

  const props: OperationCardProps = {
    title: properties.summary,
    category: properties.category,
    iconUrl: properties.api.iconUri,
    id: operation.id,
    connectorName: properties.api.displayName,
    subtitle: properties.description,
  };

  it('renders operation card', () => {
    const component = renderer.render(<OperationCard {...props}></OperationCard>);
    expect(component).toMatchSnapshot();
  });
});
