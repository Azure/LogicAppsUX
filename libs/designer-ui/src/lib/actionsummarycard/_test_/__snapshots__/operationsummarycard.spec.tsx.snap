// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recommendation panel renders operation card 1`] = `
<div
  className="msla-operation-card"
>
  <div>
    <div
      className="msla-card-title-container"
    >
      <Memo(Image)
        alt="logo for Azure Blob Storage"
        className="msla-card-logo"
        imageFit={1}
        src="https://connectoricons-prod.azureedge.net/releases/v1.0.1563/1.0.1563.2732/azureblob/icon.png"
      />
    </div>
    <div
      className="msla-card-title-container"
    >
      <Text
        className="msla-card-title"
      >
        When a blob is added or modified (properties only) (V2)
      </Text>
      <button
        onClick={[Function]}
      >
        Select
      </button>
    </div>
  </div>
  <Text
    className="msla-card-description"
  >
    This operation triggers a flow when one or more blobs are added or modified in a container. This trigger will only fetch the file metadata. To get the file content, you can use the \\"Get file content\\" operation. The trigger does not fire if a file is added/updated in a subfolder. If it is required to trigger on subfolders, multiple triggers should be created.
  </Text>
  <div
    className="msla-tag-container"
  >
    <Text
      className="msla-tag"
    >
      Azure Blob Storage
    </Text>
    <Text
      className="msla-tag"
    >
      Trigger
    </Text>
    <Text
      className="msla-tag"
    >
      Azure
    </Text>
  </div>
</div>
`;
